api/renter.go

```diff
@@ -22,7 +22,7 @@ var (
 	// contracts with if the value is not specified explicity in the call to
 	// SetSettings.
 	recommendedHosts = build.Select(build.Var{
-		Standard: uint64(42),
+		Standard: uint64(50),
```

Comment:
note the upgrade to using 50 hosts. The long-term plan is to top at 50 hosts. At 82 hosts visible to the network, we're basically on the fringe territory of safe. We can afford up to 32 hosts who are too expensive or otherwise being dumb. Sia is on a good growth trajectory though, and we've been above 70 hosts for a very long time.

I think it's fine.

42 -> 50 means that we have at all times an extra 8 hosts to use for speedboosts, etc. File redundancy was also changed to using 30 total hosts instead of 32, so it means that the final 1x redundancy will always be able to use at least 20x parallelism. It will be the 20 slowest hosts, but it's still 20 hosts. Previously it was just 10. And further, of those 10, we were seeing 8 completely unusable in the common case. So really it was only 3 hosts in use in the common case. Here we should see that number go from 3 -> 12 in the common case.

This PR also fixes some of the bugs that were causing hosts to be unreachable (can't use a host that's not in the hostdb). So that '8 in the common case' should also be going down automatically.

Long story short, this combined with a number of other changes should mean that people don't have problems anymore seeing slow uploads on the tail of a file.